game-core/src/main/java/games/strategy/triplea/settings/ClientSetting.java

```diff
@@ -205,8 +205,12 @@ public static void showSettingsWindow() {
    * values saved values until after 'flush' has been called.
    */
   public static void flush() {
+    flush(getPreferences());
```

Comment:
'flush' is pretty expensive, I think we probably should always do it on a background thread. The 'save' sounds menu used to be very slow to close (5~6 seconds), because it was saving each preference with a flush one by one. IIRC flush is about 100~250ms. Getting that off of EDT always would be a win.